ros2bag_tools/ros2bag_tools/verb/video.py

```diff
@@ -202,21 +212,34 @@ def main(self, *, args):  # noqa: D102
                 continue
             if cut_result == FilterResult.STOP_CURRENT_BAG:
                 break
-            cv_image = image_bridge.imgmsg_to_cv2(image, args.encoding)
+
+            if is_compressed:
+                cv_image = image_bridge.compressed_imgmsg_to_cv2(image)
+            else:
+                cv_image = image_bridge.imgmsg_to_cv2(image, args.encoding)
             if args.image_resize:
                 width = int(cv_image.shape[1] * args.image_resize)
                 height = int(cv_image.shape[0] * args.image_resize)
                 dim = (width, height)
                 cv_image = cv2.resize(
                     cv_image, dim, interpolation=RESIZE_INTERPOLATION)
 
-            cv_image = cvtColorForDisplay(
-                cv_image, image.encoding, args.encoding,
-                do_dynamic_scaling=args.do_dynamic_scaling,
-                min_image_value=args.min_image_value or 0.0,
-                max_image_value=args.max_image_value or 0.0,
-                # colormap=args.colormap
-            )
+            if is_compressed:
+                cv_image = cvtColorForDisplay(
```

Comment:
In particular, I would appreciate some feedback on how to properly handle the compressed image encoding.

Reply:
Thank you for the contribution!

Regarding compressed image encoding, you don't need to handle cvtColorForDisplay differently.
But to support all `CompressedImage` messages you need to get the encoding from the `format` field instead of the `encoding` field in `Image`. [This line in exporter/image.py](https://github.com/Kettenhoax/ros2bag_tools/blob/master/ros2bag_tools/ros2bag_tools/exporter/image.py#L236) shows the usage of a helper function that can parse the encoding.
Can you update the PR to use this function?

In addition, please run automated code style checks, which helps us stick with the ros code conventions.
Running this command in your workspace will show you the warnings:

```bash
colcon test --packages-select ros2bag_tools --event-handlers console_cohesion+
```

Reply:
Yes, sorry I will try and address the comments later today!

Reply:
@Kettenhoax addressed! Please let me know if there is anything else.